### ISSUE TITLE
MediaElement.js upgrade analysis

### ISSUE THREAD

**Issue by swashbuck** [adaptlearning/adapt-contrib-media#298]
### Subject of the enhancement
Explore updating the MediaElement.js library. The primary goal is to fix some existing a11y issues like #297 and #232. It is unclear if updating the library will fix those. New features should not be considered as part of this analysis.

### Your environment
* Media 6.7.0

**Comment by swashbuck** [comment]
From MediaElement.js version 3.x and later, the [namespace](https://github.com/mediaelement/mediaelement/blob/master/MIGRATION.md#migrating-from-2x-to-4x-version) has changed from `mejs-` to `mejs__`. There is an option to use the legacy namespace. However, the Vanilla styles would still need to be updated to support the new version. This means that users who update the Vanilla theme will be forced to also update the Media plugin in their courses.

If we switch to using the new namespace, we can leave legacy styles in the Vanilla theme. So, older versions of the Media component would be still supported by the Vanilla theme. The downside is that custom plugins that extend the Media plugin may break due to the namespace change. However, I think this is the better option.

Thoughts?

**Comment by oliverfoster** [comment]
How far have you been down either option?

**Comment by swashbuck** [comment]
> How far have you been down either option?

I've been using the new namespaces `mejs__`. However, it wouldn't be too much work to back out of it. Mostly a find and replace chore.

**Comment by oliverfoster** [comment]
I think carrying on with the new `mejs__` is probably the sensible option. we'll have to do it at some point. Now seems good. 

Are the styles very similar? Do you have any sense of the work required to update the library and keep the legacy css namespace?

**Comment by swashbuck** [comment]
Yeah, that's what I was thinking. They may eventually remove the `classPrefix` option entirely, so might as well transition now.

I'll be looking into the theme styles currently. Just resolved an issue that was blocking me.

**Comment by swashbuck** [comment]
Should we **remove Vimeo support** since there is a dedicated [plugin](https://github.com/adaptlearning/adapt-vimeo) for Vimeo? This would remove some complexity since [froogaloop.js](https://github.com/dilipsam/froogaloop) is used solely for Vimeo afaik.

**Comment by oliverfoster** [comment]
And anything YouTube related for the same reason.
